actionview/lib/action_view/helpers/asset_tag_helper.rb

```diff
@@ -364,6 +375,16 @@ def multiple_sources_tag_builder(type, sources)
           end
         end
 
+        def resolve_image_source(source, skip_pipeline)
+          if source.is_a?(Symbol) || source.is_a?(String)
```

Comment:
I'd invert this conditional. I'm sure we can check if source respond to a method like `to_model` before calling `url_for`.

Reply:
@colorfulfool @rafaelfranca Fwiw, this broke our use of CarrierWave. Prior to this change, we could pass the thumbnail directly into the `image_tag` method:

```ruby
<%= image_tag(user.avatar.tiny) %>
```

After this change, it throws `Can't resolve image into URL: undefined method `to_model' for #<AvatarUploader...`

Adding `.url` fixes it, and it's arguable that we were using CarrierWave wrong, but passing an object that gets `to_s` called on it seems pretty common. I was surprised that this broke on a minor point release.

Do you think it's worth changing to support that behavior, or should we keep it as is?